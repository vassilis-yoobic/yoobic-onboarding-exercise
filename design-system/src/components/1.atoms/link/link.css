.link {
  color: inherit !important;
  text-decoration: none !important;
}
/*# sourceMappingURL=link.css.map */