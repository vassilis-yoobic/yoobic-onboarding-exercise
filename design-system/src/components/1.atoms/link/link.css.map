{
    "version": 3,
    "mappings": "AAAA,AAAA,KAAK,CAAC;EACJ,KAAK,EAAE,kBAAkB;EACzB,eAAe,EAAE,eAAe;CACjC",
    "sources": [
        "link.scss"
    ],
    "names": [],
    "file": "link.css"
}