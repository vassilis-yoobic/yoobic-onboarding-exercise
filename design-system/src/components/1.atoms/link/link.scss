.link {
  color: inherit !important;
  text-decoration: none !important;
}
