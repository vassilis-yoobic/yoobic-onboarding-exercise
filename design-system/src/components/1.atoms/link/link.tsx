import { Component, ComponentInterface, h, Host, Prop } from '@stencil/core';

@Component({
  tag: 'cu-link',
  styleUrl: 'link.css',
})

export class CustomLink implements ComponentInterface {
  @Prop() href!: string;
  @Prop() icon?: string;

  render() {
    return (
      <Host>
        <a href={`${this.href}`} class="link">
          <slot/>
        </a>
      </Host>);
  }
}
