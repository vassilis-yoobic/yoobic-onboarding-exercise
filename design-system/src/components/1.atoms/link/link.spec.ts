import { newSpecPage } from "@stencil/core/testing";

import { CustomLink } from "./link";

describe('1.atoms/link', () => {
  describe('rendering', () => {
    let page;

    beforeEach(async() => {
      page = await newSpecPage({
        components: [CustomLink],
        html: '<cu-link href="test">Test</cu-link>'
      })
    })

    describe('props', () => {
      it('should render', async () => {
        expect(page.root).toMatchSnapshot();
      })
      it('should render image only', async () => {
        expect(page.root).toMatchSnapshot();
      })
    })
  })
})
