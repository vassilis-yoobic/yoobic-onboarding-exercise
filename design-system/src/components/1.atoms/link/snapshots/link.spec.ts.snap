exports[`1.atoms/link rendering props should render`] = `
<cu-link>
  <a class="link" href="#">
    Test
    <slot />
  </a>
</cu-link>
`;

exports[`1.atoms/link rendering props should render image only`] = `
<cu-link>
  <a class="link" href="#">
    <st-image>
      <img class="img Small" src="img.png" alt="" />
    </st-image>
  </a>
</cu-link>
`;
